lib/opentelemetry-node.js

```diff
@@ -348,14 +352,25 @@ module.exports = function (RED) {
       logEvent(node, '4.postDeliver', sendEvent)
       const span = createSpan(tracer, sendEvent.msg, sendEvent.destination.node, node, ignoredTypesList.includes(sendEvent.destination.node.type))
       if (propagateHeadersTypesList.includes(sendEvent.destination.node.type)) {
-        // add trace context in http request headers
-        const output = {}
-        const ctx = trace.setSpan(context.active(), span)
-        propagation.inject(ctx, output)
-        if (!sendEvent.msg.headers) {
-          sendEvent.msg.headers = {}
+        const output = {};
+        const ctx = trace.setSpan(context.active(), span);
+        propagation.inject(ctx, output);
+        switch (sendEvent.destination.node.type) {
+          // add trace context in mqtt v5 user properties
+          case "mqtt out":
+            if (!sendEvent.msg.userProperties) {
+              sendEvent.msg.userProperties = {};
+            }
+            Object.assign(sendEvent.msg.userProperties, output);
+            break;
+          default:
```

Comment:
A switch for only one case, but it's ok to avoid too much if/else. Was it your idea?

Reply:
yes, here one has to also add  "amqp-out" case.